### ISSUE TITLE
Will there ever be a 1.3.0 release?

### ISSUE THREAD

**Issue by Quintus** [datamapper/dm-core#257]
The title basically says all. Last commit on `master` is about a year old, the 1.3.0 milestone is due for over a year now. Does it even make sense to build a web application relying on DataMapper now? I’ve always preferred DataMapper to ActiveRecord, but given the stalled development this does not seem a good choice anymore.

Valete.

**Comment by martinemde** [comment]
I just noticed ardm was mentioned here. We're still actively developing ardm because our main app is locked in to DataMapper (we've hacked up to rails4 but otherwise we're in the same boat as many others).

My hope is that we can almost completely shim datamapper on top of active record. We've had a lot of success so far. I urge you to contact me directly or file issues on engineyard/ardm (even if they're just "wtf?" issues) and I'll do my best to answer them and improve documentation.

**Comment by basex** [comment]
I noticed some commits on the last days. So maybe we will have a a 1.3.0 version after all :)
https://github.com/datamapper/dm-core/blob/master/lib/dm-core/version.rb

**Comment by solnic** [comment]
@basex yeah we're trying to get crucial gems to a releasable state. I already managed to merge release-1.2 branch to master in dm-core and got rid off unfinished virtus integration. The build is passing on MRI 1.9-2.2 and rbx/jruby. There's still quite some work to be done in dm-types to merge release-1.2 into master and make sure all specs are passing but I'll get to it soon. I don't have much time for DM but I should be able to get dm-core, dm-types and dm-migrations 1.3.0 to a releasable state. If people can help with the rest we could push 1.3.0 release.

Regarding migration path to ROM, things started to clarify. I'm working on a [rom-dm](https://github.com/rom-rb/rom-dm) adapter and slowly porting an app from rails/dm to rails/rom. The adapter simply allows you to use DM as a backend for ROM (which is a natural way of using ROM so it's not any kind of a nasty hack or anything).

**Comment by martinemde** [comment]
As far as fixing all the `dm-*` projects for release, I've already done that here for most of them: https://github.cob/ar-dm/

I did take some liberties in making this all work. For example I chopped out a few ruby versions, but they could be added back in. For the most part the `ardm-*` are very honest attempts to get all of the pending work on datamapper merged and released. Engine Yard Cloud is currently running on these versions and they pass on all rubies through 2.1.5 (and the tests are the only reason dm-core isn't working on 2.2, the code itself works fine as do most of the dm- projects).

If you'd like my help porting these over as dm- releases, let me know. I also suggest anyone who wants to help to try changing the names of `dm-*` gems to `ardm-*` in your gemfile and rebundle. If all tests pass then try again with the versions loosened to `~> 1.2` to grab newer versions that I've released since.

**Comment by solnic** [comment]
That's good to know. What I did, which was important, was reverting
half-baked virtus integration. Then we merged release-1.2 branches into
master in most important projects.

We should probably look at ar-dm changes and port them to main projects if
it makes sense.

I guess we could push 1.3.0 of main projects (core, migrations, types,
validations and maybe a couple more) once all specs are passing and a bunch
of people confirmed it all works in their apps :)

**Comment by robclancy** [comment]
It's great working with dependencies with bugs that get fixed but you can't reliably get them into the project...

**Comment by basex** [comment]
I can make the tests work in some of the dm gems.
Can someone approve the pull requests?

**Comment by solnic** [comment]
@basex that would be lovely

**Comment by martinemde** [comment]
I started pushing PRs with the changes needed to modernize the dm gems based on ardm-* for [dm-active_model](https://github.com/datamapper/dm-active_model/pull/5) and [dm-aggregates](https://github.com/datamapper/dm-aggregates/pull/21) so far. The process is fairly simple. Pull down datamapper version at master, grab my ardm- version for that gem, then start with dm master version and rebase ardm/master version on top and fix the conflicts (usually based around changes I made for ardm naming). Then make sure all the ardm references are removed as you go, push and PR. I'll do more when I have a few more minutes. It takes about 20-30 minutes per gem all together.

Another thing of note is that I was able to break the strict dependencies between version of datamapper that required lock-stop versioning. For example, dm-aggregates works fine against dm-core 1.2 or (my version) of dm-core 1.3. We don't have to have such strict versioning so these newer versions can be released as they come along, not all at the same time.

**Comment by tpitale** [comment]
I'm currently in the process of attempting to close PRs and issues in preparation for a dm-core 1.3.0. Any help from anyone on this thread would be much appreciated!
